config/elastic-apm-laravel.php

```diff
@@ -13,31 +13,36 @@
 
     'app' => [
         // The app name that will identify your app in Kibana / Elastic APM, limited characters allowed
-        'appName' => preg_replace('/[^a-zA-Z0-9 _-]/', '-', env('APM_APPNAME')),
+        'appName' => preg_replace('/[^a-zA-Z0-9 _-]/', '-', env('ELASTIC_APM_SERVICE_NAME', env('APM_APPNAME', 'Laravel'))),
 
         // The version of your app
-        'appVersion' => env('APM_APPVERSION'),
+        'appVersion' => env('ELASTIC_APM_SERVICE_VERSION', env('APM_APPVERSION')),
     ],
 
     'env' => [
         // whitelist environment variables OR send everything
         'env' => ['DOCUMENT_ROOT', 'REMOTE_ADDR'],
 
         // Application environment
-        'environment' => env('APM_ENVIRONMENT'),
+        'environment' => env('APM_ENVIRONMENT', 'development'),
     ],
 
     'server' => [
         // The apm-server to connect to
-        'serverUrl' => env('APM_SERVERURL'),
+        'serverUrl' => env('ELASTIC_APM_SERVER_URL', env('APM_SERVERURL', 'http://127.0.0.1:8200')),
 
         // Token for x
-        'secretToken' => env('APM_SECRETTOKEN'),
+        'secretToken' => env('ELASTIC_APM_SECRET_TOKEN', env('APM_SECRETTOKEN')),
 
         // Hostname of the system the agent is running on.
         'hostname' => null,
     ],
 
+    'agent' => [
+        'hostname' => env('ELASTIC_APM_HOSTNAME', gethostname()),
```

Comment:
The `hostname` value is already defined in the `server` block, and it appears correctly in APM, under the `host.hostname` property. Which one should we keep?

Reply:
I missed that. Keeping the original seems best. I think replacing the `null` with the env value is warranted since the Agent implies you can specify the hostname with that variable. If we don't add it, then there would be no way to explicitly set it when the config is cached.

It makes sense to me to consolidate Agent settings into a specific key in the config array, which is why I added `agent` in there. I know that moving existing items would have consequences and possibly for users to updated previously published config files, so that should wait for a major release.